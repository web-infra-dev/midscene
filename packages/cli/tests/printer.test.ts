import {
  currentSpinningFrame,
  flowItemBrief,
  isTTY,
  spinnerFrames,
  spinnerInterval,
} from '@/printer';
import { describe, expect, test } from 'vitest';

describe('printer', () => {
  test('action brief text', () => {
    expect(flowItemBrief({ ai: 'search for weather' })).toMatchSnapshot();
    expect(flowItemBrief({ sleep: 1000 })).toMatchSnapshot();
    expect(
      flowItemBrief({ aiWaitFor: 'wait for something' }),
    ).toMatchSnapshot();
  });
});
