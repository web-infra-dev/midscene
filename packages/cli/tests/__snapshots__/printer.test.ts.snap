// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`printer > action brief text 1`] = `"aiAction: search for weather"`;

exports[`printer > action brief text 2`] = `"sleep: 1000"`;

exports[`printer > action brief text 3`] = `"aiWaitFor: wait for something"`;
